include/storage/io.hpp

```diff
@@ -192,7 +199,11 @@ class FileWriter
     {
         output_stream.open(filepath, std::ios::binary);
         if (!output_stream)
-            throw util::exception("Error opening " + filepath.string());
+        {
+            throw util::FileWriteException(std::string(filepath.string()) + ": " +
+                                               std::string(std::strerror(errno)),
```

Comment:
Yes, I'm aware it's not guaranteed.  Including this value is a calculated risk.  I'll change the wording to indicate that it's not necessarily the correct value, but my tests show that it is correct in the cases we care about, on current `libstdc++` implementations.

Something like `Error opening filename.osrm: Possibly caused by Permission Denied`.